Mathlib/Combinatorics/SimpleGraph/Circulant.lean

```diff
@@ -0,0 +1,136 @@
+/-
+Copyright (c) 2024 Iván Renison, Bhavik Mehta. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Iván Renison, Bhavik Mehta
+-/
+import Mathlib.Combinatorics.SimpleGraph.Hasse
+import Mathlib.Data.Set.Pointwise.Basic
+
+/-!
+# Definition of circulants graphs
+
+This file defines and proves several fact about circulants graphs.
+A circulant graph over type `G` with jumps `s : Set G` is a graph in which two vertices `u` and `v`
+are adjacent if and only if `u - v ∈ s` or `v - u ∈ s`. The elements of `s` are called jumps.
+
+## Main declarations
+
+* `SimpleGraph.CirculantGraph s`: the circulant graph over `G` with jumps `s`.
+* `SimpleGraph.cycleGraph n`: the cycle graph over `Fin n`.
+-/
+
+namespace SimpleGraph
+
+/-- Circulant graph over additive group `G` with jumps `s` -/
+@[simps!]
+def CirculantGraph {G : Type*} [AddGroup G] (s : Set G) : SimpleGraph G :=
+  fromRel (· - · ∈ s)
+
+variable {G : Type*} [AddGroup G] (s : Set G)
+
+theorem circulantGraph_eq_erase_zero : CirculantGraph s = CirculantGraph (s \ {0}) := by
+  ext (u v : G)
+  simp only [CirculantGraph, fromRel_adj, and_congr_right_iff]
+  intro (h : u ≠ v)
+  apply Iff.intro
+  · intro h1
+    cases h1 with
+      | inl h1 => exact Or.inl ⟨h1, sub_ne_zero_of_ne h⟩
+      | inr h1 => exact Or.inr ⟨h1, sub_ne_zero_of_ne h.symm⟩
+  · intro h1
+    cases h1 with
+      | inl h1 => exact Or.inl h1.left
+      | inr h1 => exact Or.inr h1.left
+
+theorem circulantGraph_eq_symm : CirculantGraph s = CirculantGraph (s ∪ (-s)) := by
+  ext (u v : G)
+  simp only [CirculantGraph, fromRel_adj, Set.mem_union, Set.mem_neg, neg_sub, and_congr_right_iff,
+    iff_self_or]
+  intro _ h
+  exact Or.symm h
+
+instance [DecidableEq G] [DecidablePred (· ∈ s)] : DecidableRel (CirculantGraph s).Adj :=
+  fun _ _ => inferInstanceAs (Decidable (_ ∧ _))
+
+theorem circulantGraph_adj_translate {s : Set G} {u v d : G} :
+    (CirculantGraph s).Adj (u + d) (v + d) ↔ (CirculantGraph s).Adj u v := by simp
+
+/-- Cycle graph over `Fin n` -/
+def cycleGraph : (n : ℕ) → SimpleGraph (Fin n)
```

Comment:
```suggestion
def CycleGraph : (n : ℕ) → SimpleGraph (Fin n)
```

Reply:
I used `cycleGraph` for using the same style that `pathGraph`, you think that `CycleGraph` is better?

Reply:
Also, `completeGraph`, `emptyGraph`, `completeBipartiteGraph`, `hasse` are in lower case, so I will change `CirculantGraph`